### PR TITLE
Update test cases generated by make_test() method to support running them in script mode.

### PR DESCRIPTION
**Pull request by fatcat-z**
Update code relative to all test_gru_* test cases. Still disable them running in script mode because of 2 issues:

Operator aten::as_tensor is not supported by exporter yet.
      - https://msdata.visualstudio.com/Vienna/_workitems/edit/1055382
Operator aten::_pack_padded_sequence is not supported by exporter yet.
      - https://msdata.visualstudio.com/Vienna/_workitems/edit/1055384

### CODE REVIEW

File: test/onnx/model_defs/rnn_model_with_packed_sequence.py [hunk 1]
```diff
@@ -15,3 +15,29 @@ def forward(self, input, *args):
         ret, rets = rets[0], rets[1:]
         ret, _ = rnn_utils.pad_packed_sequence(ret, self.batch_first)
         return tuple([ret] + list(rets))
+
```
Comment (neginraoof) [inline test/onnx/model_defs/rnn_model_with_packed_sequence.py:18]: Do we still need to keep RnnModelWithPackedSequence?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -6903,7 +7003,13 @@ def make_test(name, base, layer, bidirectional, initial_state,
 
     # Cannot export with older opsets because of 'ConstantFill' op
     # ConstantFill was a temp op removed at opset 8. This is no longer supported by onnxruntime
-    @disableScriptTest()  # Test code not scriptable
+    # There are still some issues prevent us from enabling script test for these scenarios:
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:7006]: Thanks a lot for there comments!

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 3]
```diff
@@ -5975,13 +6012,76 @@ def make_input(batch_size):
...
+
+        class GRUNoSeqLengthWithStateModel(torch.nn.Module):
+            def __init__(self, layers, bidirect, dropout, batch_first):
+                super(GRUNoSeqLengthWithStateModel, self).__init__()
+                self.batch_first = batch_first
+                self.inner_model = torch.nn.GRU(RNN_INPUT_SIZE, RNN_HIDDEN_SIZE, num_layers=layers, 
+                                                bidirectional=bidirectional, dropout=dropout,
+                                                batch_first=batch_first)
+
+            def forward(self, input, hx):
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:6058]: Do we need special annotations for input here? Or is it a regular Tensor?

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 4]
```diff
@@ -5975,13 +6012,76 @@ def make_input(batch_size):
...
+
+        class GRUNoSeqLengthWithoutStateModel(torch.nn.Module):
+            def __init__(self, layers, bidirect, dropout, batch_first):
+                super(GRUNoSeqLengthWithoutStateModel, self).__init__()
+                self.batch_first = batch_first
+                self.inner_model = torch.nn.GRU(RNN_INPUT_SIZE, RNN_HIDDEN_SIZE, num_layers=layers, 
+                                                bidirectional=bidirectional, dropout=dropout,
+                                                batch_first=batch_first)
+
+            def forward(self, input):
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:6047]: Do we need special annotations for input here? Or is it a regular Tensor?

**Review by neginraoof — COMMENTED** [review]
Looks good overall. Left a few comments.
Thanks!

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 3 above)
Comment (fatcat-z) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:6058]: It's just a regular Tensor which would be inferred by Python, so we don't need an annotation.

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 4 above)
Comment (fatcat-z) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:6047]: It's just a regular Tensor which would be inferred by Python, so we don't need an annotation.

File: test/onnx/model_defs/rnn_model_with_packed_sequence.py (hunk 1 above)
Comment (fatcat-z) [reply test/onnx/model_defs/rnn_model_with_packed_sequence.py:18]: I see it was used in test_pytorch_onnx_caffe2.py so I'd like to keep it there.

**Review by neginraoof — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**